### PR TITLE
[THORN-2116] Adding the info about the project stages

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

**Comment by sberyozkin** [comment]
Hi Sheldon, what I'm not sure is how to escape "---" sequences within the YAML file, please see the attached image to THORN-2116.

**Comment by kenfinnigan** [comment]
Is this content the same/similar to https://docs.thorntail.io/2.3.0.Final-SNAPSHOT/#default-thorntail-yaml-files_thorntail and https://docs.thorntail.io/2.3.0.Final-SNAPSHOT/#non-default-thorntail-yaml-configuration-files_thorntail

File: docs/modules/con_thorntail-yaml-project-stages.adoc [hunk 1]
```diff
@@ -0,0 +1,80 @@
+
+[id='the-thorntail-yaml-project-stages_{context}']
+= Configuration overlays using stage properties
+
+In some cases it may be desirable to extract some of the configuration properties and switch them depending on the target environment. Project stages are a way to move environment specific configuration properties into an external file.
```
Comment (Ladicek) [inline docs/modules/con_thorntail-yaml-project-stages.adoc]: "switch _between_ them"?

Also I think the convention is 1 sentence = 1 line.

File: docs/modules/con_thorntail-yaml-project-stages.adoc [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+.Stage configurations in project-defaults.yml
+====
+This example shows three stages, `default`, `development` and `production` being configured in the default project-defaults.yml.
```
Comment (Ladicek) [inline docs/modules/con_thorntail-yaml-project-stages.adoc:9]: Do we still want to encourage this? I'm not even sure it's tested :-)

**Comment by tradej** [comment]
I will wait with my review for when the technical questions are answered.

**Comment by sberyozkin** [comment]
Ken, re https://docs.thorntail.io/2.3.0.Final-SNAPSHOT/#default-thorntail-yaml-files_thorntail, to me it looks like the new content is different, as it shows how the project-default.xml can have overlaying configuration parts with only one of them activated, 'default', if no -Dswarm.project.stage is provided, or the chosen stage.
I'm less certain about https://docs.thorntail.io/2.3.0.Final-SNAPSHOT/#non-default-thorntail-yaml-configuration-files_thorntail though. It looks like it implies that multiple configuration files can be used at the same time, with this line "The first file containing a given configuration item takes precedence over others" - to me it reads that all the files are consumed but the ordering is applied ?
The new content implies that if -Dswarm.project.stage.file is provided then only a single file is loaded...

File: docs/modules/con_thorntail-yaml-project-stages.adoc (hunk 1 above)
Comment (sberyozkin) [reply docs/modules/con_thorntail-yaml-project-stages.adoc]: It was a copy and paste from the old content, will update...

Comment (sberyozkin) [reply docs/modules/con_thorntail-yaml-project-stages.adoc]: I've kept a single sentence only...I'm not sure, when I read it again and again what the above sentence about the project stages means :-)

File: docs/modules/con_thorntail-yaml-project-stages.adoc (hunk 2 above)
Comment (sberyozkin) [reply docs/modules/con_thorntail-yaml-project-stages.adoc:9]: I guess having a single file overlaying the properties may be reasonable for a start, for the users to get a quick try at this approach before starting creating multiple files...

File: docs/modules/con_thorntail-yaml-project-stages.adoc (hunk 1 above)
Comment (Ladicek) [reply docs/modules/con_thorntail-yaml-project-stages.adoc]: Re the sentences, sorry I wasn't clear -- it's an docs authoring convention, to never wrap sentences in the middle and to never have more than one sentence on one line.

**Comment by Ladicek** [comment]
Just looked at the links Ken mentioned -- yea, that's my concern as well. The text added in this PR describes an old style of using project stages, while https://docs.thorntail.io/2.2.0.Final/#configuring-a-thorntail-application-using-yaml-files_thorntail describes a new style that should be preferred. I'm not sure how much the old style still works -- some of it certainly does (the `swarm.project.stage` and `swarm.project.stage.file` system properties), but e.g. the text about multiple stages in a single YAML file, that should certainly be considered deprecated.

**Comment by sberyozkin** [comment]
Sure, no problems, I thought it was a different content. I'll let you decide if this PR should be closed or not.

**Comment by sberyozkin** [comment]
Ladislav, Ken, I will just close this PR for now, OK ?

**Comment by Ladicek** [comment]
Makes sense to me.

**Comment by sberyozkin** [comment]
Sure, closing for now, will re-open if needed

**Pull request closed without merging**